packages/telemetry/src/make-slog-sender.js

```diff
@@ -1,32 +1,121 @@
   }
 
-  if (slogSenderModule.startsWith('.')) {
-    // Resolve relative to the current working directory.
-    slogSenderModule = path.resolve(slogSenderModule);
+  if (!slogSenderModules.length) {
+    return undefined;
   }
 
-  console.warn(`Loading makeSlogSender from ${slogSenderModule}`);
-  const { makeSlogSender: maker } = await import(slogSenderModule);
-  if (typeof maker !== 'function') {
-    throw Error(`${slogSenderModule} did not export a makeSlogSender function`);
+  console.warn(
+    `Loading slog sender${
+      slogSenderModules.length > 1 ? 's' : ''
+    } from ${slogSenderModules.join(', ')}`,
+  );
+
+  const makersInfo = await Promise.all(
+    slogSenderModules.map(async moduleIdentifier =>
+      import(moduleIdentifier)
+        .then(
+          /** @param {{makeSlogSender: (opts: {}) => Promise<SlogSender | undefined>}} module */ ({
+            makeSlogSender: maker,
+          }) =>
+            typeof maker === 'function'
+              ? { maker, moduleIdentifier }
+              : Promise.reject(
+                  new Error(`No 'makeSlogSender' function exported by module`),
+                ),
+        )
+        .catch(err => {
+          console.warn(
+            `Failed to load slog sender from ${moduleIdentifier}.`,
+            err,
+          );
+          return undefined;
+        }),
+    ),
+  );
+
+  if (!makersInfo.filter(Boolean).length) {
+    return undefined;
   }
 
-  const { stateDir = tmp.dirSync().name } = makerOpts;
-  if (stateDir !== makerOpts.stateDir) {
+  let stateDir = stateDirOption;
+
+  if (stateDir === undefined) {
+    stateDir = tmp.dirSync().name;
     console.warn(`Using ${stateDir} for stateDir`);
   }
-  return maker({ ...makerOpts, stateDir });
+
+  const senders = await Promise.all(
+    makersInfo.map(async makerInfo => {
+      if (!makerInfo || makerInfo.maker === makeSlogSender) {
+        return undefined;
+      }
+      const { maker, moduleIdentifier } = makerInfo;
+      return maker({
+        ...otherOpts,
+        stateDir,
+        env: { SLOGSENDER: moduleIdentifier, ...otherEnv },
+      });
+    }),
+  ).then(
+    potentialSenders =>
+      /** @type {SlogSender[]} */ (potentialSenders.filter(Boolean)),
+  );
+
+  if (!senders.length) {
+    return undefined;
+  } else if (senders.length === 1) {
+    return senders[0];
+  } else {
+    const hasSenderNeedingJsonObj = senders.some(sender => sender.length === 2);
+    const getJsonObj = hasSenderNeedingJsonObj
+      ? serializeSlogObj
+      : () => undefined;
+    /** @type {SlogSender} */
+    const slogSender = (slogObj, jsonObj = getJsonObj(slogObj)) => {
+      for (const sender of senders) {
+        try {
+          sender(slogObj, jsonObj);
+        } catch (err) {
+          console.error('WARNING: slog sender error', err);
+        }
+      }
+    };
+    if (hasSenderNeedingJsonObj) {
+      Object.defineProperty(slogSender, 'length', { value: 2 });
+    }
```

Comment:
Playing with the function arity idiom is pretty unclear.

Could you instead use `Object.assign(slogSender, { needsJsonObject: true })`, and do the same for the other slogSenders?